src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -5265,4 +5265,1285 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
+        {
+            Rpp8u *srcPtrRow, *dstPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB, *srcPtrTopRow, *srcPtrBottomRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRow = dstPtrChannel;
+            if(dstDescPtr->layout == RpptLayout::NCHW)
+            {
+                dstPtrRowR = dstPtrRow;
+                dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+                dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+            }
+            for(int i = 0; i < dstImgSize[batchCount].height; i++)
+            {
+                compute_resize_src_loc(i, hRatio, heightLimit, srcLocationRowFloor, weightedHeight);
+                weightedHeight1 = 1 - weightedHeight;
+                srcPtrTopRow = srcPtrRow + srcLocationRowFloor * srcBufferLength;
+                srcPtrBottomRow  = srcPtrTopRow + srcBufferLength;
+                int vectorLoopCount = 0;
+                __m128 pWeightedHeight = _mm_set1_ps(weightedHeight);
+                __m128 pWeightedHeight1  = _mm_set1_ps(weightedHeight1);
+                __m128 pRow[12], pPixels[3];
+
+                for (; vectorLoopCount < alignedLength; vectorLoopCount+=4)
+                {
+                    pDstLoc = _mm_setr_ps(vectorLoopCount, vectorLoopCount + 1, vectorLoopCount + 2, vectorLoopCount + 3);
+                    compute_resize_src_loc_sse(pDstLoc, pWRatio, pWidthLimit, srcLocCF, pWeightedWidth, true);
+                    pWeightedWidth1  = _mm_sub_ps(pOne, pWeightedWidth);
+                    compute_bilinear_coefficients_sse(pWeightedWidth, pWeightedWidth1, pWeightedHeight, pWeightedHeight1, pBilinearCoeff);
+                    rpp_simd_load(rpp_bilinear_load4_u8pkd3_to_f32pln3, srcPtrTopRow, srcPtrBottomRow, srcLocCF, pRow);
+                    compute_bilinear_interpolation_sse(pRow, pBilinearCoeff, pPixels);
+                    if(dstDescPtr->layout == RpptLayout::NCHW)
+                    {
+                        rpp_simd_store(rpp_store4_f32pln3_to_u8pln3, dstPtrRowR, dstPtrRowG, dstPtrRowB, pPixels);
+                        dstPtrRowR += 4;
+                        dstPtrRowG += 4;
+                        dstPtrRowB += 4;
+                    }
+                    else
+                    {
+                        rpp_simd_store(rpp_store4_f32pln3_to_u8pkd3, dstPtrRow, pPixels);
+                        dstPtrRow += 12;
+                    }
+                }
+                if(dstDescPtr->layout == RpptLayout::NCHW)
+                {
+                    for (; vectorLoopCount < dstImgSize[batchCount].width; vectorLoopCount++)
+                    {
+                        compute_resize_src_loc(vectorLoopCount, wRatio, widthLimit, srcLocationColumnFloor, weightedWidth, true);
+                        weightedWidth1 = 1 - weightedWidth;
+                        Rpp32f m1 = weightedHeight1 * weightedWidth1;
```

Comment:
You could have a templated compute_bilinear_interpolation() just like the compute_bilinear_interpolation_sse() for L5392-L5407 and reuse everywhere